### PR TITLE
Exclude CI job 8.10.7 on Windows that corrupts all CI

### PR DESCRIPTION
**Pull request by Mikolaj**
See https://github.com/haskell/cabal/issues/8356 and https://github.com/haskell/cabal/pull/8385.

### CODE REVIEW

**Comment by Mikolaj** [comment]
The symptoms are so strange and mysterious, that this is at best a workaround of https://github.com/haskell/cabal/issues/8356. It would be great to get to the bottom of it, but OTOH CI needs to mostly work or dev is stalled and we accumulate bugs.

**Comment by Mikolaj** [comment]
CI passes after disabling 8.10.7 on Windows. @jneira: you might have had a good hunch that was the culprit.

Now let me try to disable the previous workaround, contained in #8357.

**Review by ulysses4ever — APPROVED** [review]

**Comment by Mikolaj** [comment]
@mergify backport 3.8

**Comment by Mikolaj** [comment]
Emergency-merging before the 2-day period elapses.

**Comment by Mikolaj** [comment]
@mergify rebase

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -40,6 +40,9 @@ jobs:
         os: ["ubuntu-latest", "macos-latest", "windows-latest"]
         ghc: ["9.2.3", "9.0.2", "8.10.7", "8.8.4", "8.6.5", "8.4.4"]
         exclude:
+          # corrupts GHA cache or the fabric of reality itself, see https://github.com/haskell/cabal/issues/8356
```
Comment (jneira) [inline .github/workflows/validate.yml:43]: 🤣

**Comment by jneira** [comment]
hmm that is unfortunate, 8.10.7 is and will be a very stable version, i think it will be the succesor of 8.6.5
but i understand it should be done temporay (i hope!)

**Pull request merged**